core/src/main/java/org/infinispan/util/concurrent/locks/impl/DefaultPendingLockManager.java

```diff
@@ -50,9 +54,12 @@ public class DefaultPendingLockManager implements PendingLockManager {
 
    private static final Log log = LogFactory.getLog(DefaultPendingLockManager.class);
    private static final int NO_PENDING_CHECK = -2;
+   private final Set<PendingLockPromise> pendingPromises = ConcurrentHashMap.newKeySet();
```

Comment:
Necessary when the transactions don't go through `DefaultLockManager` to acquire the locks. We store the promises (similar to the PlaceHolderLock) so we can cancel them in case of deadlocks.